src/java/com/wolfssl/provider/jsse/WolfSSLEngineHelper.java

```diff
@@ -1121,19 +1121,43 @@ private void setLocalSigAlgorithms() {
             /* Get restricted signature algorithms for ClientHello if set by
              * user in "wolfjsse.enabledSigAlgorithms" Security property */
             String sigAlgos = WolfSSLUtil.getSignatureAlgorithms();
-
-            if (sigAlgos != null) {
-                ret = this.ssl.setSignatureAlgorithms(sigAlgos);
-                if (ret != WolfSSL.SSL_SUCCESS &&
-                    ret != WolfSSL.NOT_COMPILED_IN) {
-                    WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
-                        () -> "error restricting signature algorithms based " +
-                        "on wolfjsse.enabledSignatureAlgorithms property");
-                } else if (ret == WolfSSL.SSL_SUCCESS) {
-                    WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
-                        () -> "restricted signature algorithms based on " +
-                        "wolfjsse.enabledSignatureAlgorithms property");
-                }
+            String sigSchemes =
+                   WolfSSLUtil.getSignatureSchemes(this.clientMode);
+            String cleanSigSchemes =
+                   WolfSSLUtil.formatSigSchemes(sigAlgos, sigSchemes);
+            
+            if (cleanSigSchemes != null)
+                ret = this.ssl.setSignatureAlgorithms(cleanSigSchemes);
+            if (ret != WolfSSL.SSL_SUCCESS &&
+                ret != WolfSSL.NOT_COMPILED_IN) {
+                WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
+                    () -> "error restricting signature algorithms based " +
+                    "on \"wolfjsse.enabledSigAlgorithms\" and " +
+                    "\"jdk.tls.client.SignatureSchemes\" properties");
+            } else if (ret == WolfSSL.SSL_SUCCESS) {
+                WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
+                    () -> "restricted signature algorithms based on " +
+                    "on \"wolfjsse.enabledSigAlgorithms\" and " +
+                    "\"jdk.tls.client.SignatureSchemes\" properties");
+            }
+        }
+        else {
```

Comment:
It looks like there is good amount of duplicated code between this `if` and `else` block. Is there a way we can consolidate this into a shorter code block?

Since the difference in the `else` block is just passing in `null` to `WolfSSLUtil.formatSigSchemes()`, maybe you can just set `sigSchemes` when `this.clientMode`? ie:

```
String sigSchemes = null;
if (this.clientMode) {
    sigSchemes = WolfSSLUtil.getSignatureSchemes(this.clientMode);
}
...
cleanSigSchemes = WolfSSLUtil.formatSigSchemes(sigSchemes, sigSchemes);
```